novawallet/Common/Services/Proxy/ChainProxySyncService.swift

```diff
@@ -0,0 +1,247 @@
+    }
+
+    private func performSyncUp(
+        connection: JSONRPCEngine,
+        runtimeProvider: RuntimeCodingServiceProtocol
+    ) {
+        let storageKeyFactory = StorageKeyFactory()
+        let requestFactory = StorageRequestFactory(
+            remoteFactory: storageKeyFactory,
+            operationManager: operationManager
+        )
+        let proxyListWrapper = proxyOperationFactory.fetchProxyList(
+            requestFactory: requestFactory,
+            connection: connection,
+            runtimeProvider: runtimeProvider
+        )
+        let metaAccountsOperation = metaAccountsRepository.fetchAllOperation(with: .init())
+        let identityOperationFactory = IdentityOperationFactory(requestFactory: requestFactory)
+
+        let changesOperation = changesOperation(
+            proxyListWrapper: proxyListWrapper,
+            metaAccountsOperation: metaAccountsOperation,
+            connection: connection,
+            runtimeProvider: runtimeProvider,
+            identityOperationFactory: identityOperationFactory
+        )
+        let saveOperation = saveOperation(dependingOn: changesOperation)
+        saveOperation.addDependency(changesOperation.targetOperation)
+
+        let compoundWrapper = CompoundOperationWrapper(
+            targetOperation: saveOperation,
+            dependencies: changesOperation.allOperations
+        )
+
+        saveOperation.completionBlock = { [weak self] in
+            guard let workingQueue = self?.workingQueue, let mutex = self?.mutex else {
+                return
+            }
+
+            dispatchInConcurrent(queue: workingQueue, locking: mutex) {
+                guard self?.pendingCall === compoundWrapper else {
+                    return
+                }
+
+                self?.pendingCall = nil
+
+                do {
+                    _ = try saveOperation.extractNoCancellableResultData()
+                    self?.completeImmediate(nil)
+                } catch {
+                    self?.completeImmediate(error)
+                }
+            }
+        }
+
+        pendingCall = compoundWrapper
+        operationQueue.addOperations(compoundWrapper.allOperations, waitUntilFinished: false)
+    }
+
+    private func updatedMetaAccounts(
+        metaAccounts: [ManagedMetaAccountModel],
+        for accountId: AccountId,
+        proxieds: [ProxiedAccount],
+        identities: [AccountId: AccountIdentity]
+    ) -> [ManagedMetaAccountModel] {
+        guard metaAccounts.contains(where: { $0.info.has(accountId: accountId, chainId: chainModel.chainId) }) else {
+            return []
+        }
+        let remoteProxieds = proxieds.map {
+            ProxiedAccountModel(
+                type: $0.type,
+                accountId: $0.accountId,
+                status: .new
+            )
+        }
+        let localProxieds = metaAccounts
+            .filter { $0.info.type == .proxy && $0.info.has(accountId: accountId, chainId: chainModel.chainId) }
```

Comment:
```accountId``` here is accountId of the proxy in this case but we want to find all proxieds. So we need to check whether a given proxied wallet as a proxy in the chain account. Am I right?